components/tidb_query_expr/src/impl_json.rs

```diff
@@ -226,6 +227,16 @@ fn valid_paths(expr: &tipb::Expr) -> Result<()> {
     Ok(())
 }
 
+fn unquote_string(s: &str) -> Result<String> {
+    let first_char = s.chars().next();
+    let last_char = s.chars().nth(s.len() - 1);
```

Comment:
This seems to be not correct. len returns how many bytes are there in str, while chars is an iterator on UTF-8 characters. Please also considering adding test cases on CJK characters. After all, I suggest reusing previous logic instead of extracting quotes by ourselves.

Reply:
Yeah. I didn't notice the sematics of `len()`.